contract_compassion/model/contracts.py

```diff
@@ -0,0 +1,635 @@
+                    'will_pay_by_lsv_dd', cr)
+        else:
+            # Check if old payment_term was LSV or DD
+            for contract in self.browse(cr, uid, ids, context):
+                payment_name = contract.group_id.payment_term_id.name
+                if 'LSV' in payment_name or 'Postfinance' in payment_name:
+                    wf_service.trg_validate(
+                        uid, 'recurring.contract', contract.id,
+                        'mandate_validated', cr)
+
+    def _on_contract_lines_changed(self, cr, uid, ids, context=None):
+        """Update related invoices to reflect the changes to the contract.
+        """
+        invoice_obj = self.pool.get('account.invoice')
+        inv_line_obj = self.pool.get('account.invoice.line')
+        # Find all unpaid invoice lines after the given date
+        since_date = datetime.today().replace(day=1).strftime(DF)
+        inv_line_ids = inv_line_obj.search(
+            cr, uid, [('contract_id', 'in', ids),
+                      ('due_date', '>=', since_date),
+                      ('state', 'not in', ('paid', 'cancel'))],
+            context=context)
+        con_ids = set()
+        inv_ids = set()
+        for inv_line in inv_line_obj.browse(cr, uid, inv_line_ids, context):
+            invoice = inv_line.invoice_id
+            if invoice.id not in inv_ids or \
+                    inv_line.contract_id.id not in con_ids:
+                con_ids.add(inv_line.contract_id.id)
+                inv_ids.add(invoice.id)
+                invoice_obj.action_cancel(cr, uid, [invoice.id], context)
+                invoice_obj.action_cancel_draft(cr, uid, [invoice.id])
+                self._update_invoice_lines(cr, uid, inv_line.contract_id,
+                                           [invoice.id], context)
+        wf_service = netsvc.LocalService('workflow')
+        for invoice in invoice_obj.browse(cr, uid, list(inv_ids), context):
+            wf_service.trg_validate(
+                uid, 'account.invoice', invoice.id, 'invoice_open', cr)
+
+    def _on_group_id_changed(self, cr, uid, ids, context=None):
+        """Remove lines of open invoices and generate them again
+        """
+        self._reset_open_invoices(cr, uid, ids, context)
+        for contract in self.browse(cr, uid, ids, context=context):
+            # Update next_invoice_date of group if necessary
+            if contract.group_id.next_invoice_date:
+                next_invoice_date = datetime.strptime(
+                    contract.next_invoice_date, DF)
+                group_date = datetime.strptime(
+                    contract.group_id.next_invoice_date, DF)
+                if group_date > next_invoice_date:
+                    # This will trigger group_date computation
+                    contract.write({
+                        'next_invoice_date': contract.next_invoice_date})
+
+    def _compute_next_invoice_date(self, contract):
+        # Sponsorship case

```

Comment:
Change to docstring for commenting the method and explain we generate an invoice each month for sponsorships.